pkg/builder/controller.go

```diff
@@ -311,19 +310,16 @@ func (blder *Builder) doWatch() error {
 		return errors.New("there are no watches configured, controller will never get triggered. Use For(), Owns() or Watches() to set them up")
 	}
 	for _, w := range blder.watchesInput {
-		allPredicates := append([]predicate.Predicate(nil), blder.globalPredicates...)
```

Comment:
I liked this better, it explicitly makes sure we are not editing the `globalPredicates` value.

Reply:
My bad. You're right

Reply:
If I got it right we should also do this in l.277

Reply:
Thx for catching this!